plugin/trino-hive/src/main/java/io/trino/plugin/hive/parquet/ParquetPageSourceFactory.java

```diff
@@ -161,6 +163,12 @@ public ParquetPageSourceFactory(HdfsEnvironment hdfsEnvironment, FileFormatDataS
 
         checkArgument(acidInfo.isEmpty(), "Acid is not supported");
 
+        // per HIVE-13040 empty files are allowed
+        if (estimatedFileSize == 0) {
+            ReaderPageSource context = noProjectionAdaptation(new EmptyPageSource());
+            return Optional.of(context);
```

Comment:
is HIVE-13040 for Parquet only?

please check how this is handled in hive code. if this is for all formats,. we should handle this in generic manner as well

also, why generate a split for an empty file at all? it could be filtered out on the coordinator.

Reply:
Why do you need this if you added code in `InternalHiveSplitFactory.java `?

Reply:
Removed